internal/services/dashboard/dashboard_grafana_managed_private_endpoint_resource.go

```diff
@@ -0,0 +1,277 @@
+		},
+
+		"private_link_resource_region": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"request_message": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (r ManagedPrivateEndpointResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ManagedPrivateEndpointResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ManagedPrivateEndpointModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			client := metadata.Client.Dashboard.ManagedPrivateEndpointsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			grafanaId, err := grafanaresource.ParseGrafanaID(model.GrafanaId)
+			if err != nil {
+				return fmt.Errorf("parsing %q: %+v", model.GrafanaId, err)
+			}
+			id := managedprivateendpoints.NewManagedPrivateEndpointID(subscriptionId, grafanaId.ResourceGroupName, grafanaId.GrafanaName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			props := managedprivateendpoints.ManagedPrivateEndpointModel{
+				Location: location.Normalize(model.Location),
+				Name:     &model.Name,
+				Properties: &managedprivateendpoints.ManagedPrivateEndpointModelProperties{
+					GroupIds:                  &model.GroupIds,
+					PrivateLinkResourceId:     &model.PrivateLinkResourceId,
+					PrivateLinkResourceRegion: &model.PrivateLinkResourceRegion,
+					RequestMessage:            &model.RequestMessage,
+				},
+				Tags: &model.Tags,
+			}
+
+			if err := client.CreateThenPoll(ctx, id, props); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r ManagedPrivateEndpointResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Dashboard.ManagedPrivateEndpointsClient
+			id, err := managedprivateendpoints.ParseManagedPrivateEndpointID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			grafanaId, err := GetGrafanaIdFromManagedPrivateEndpointId(id.ID())
+			if err != nil {
+				return err
+			}
```

Comment:
I only see this function being called once and it's re-parsing an ID that's already been parsed on line 165. For consistency can we please update this to 
```suggestion
			grafanaId := grafanaresource.NewGrafanaID(id.SubscriptionId, id.ResourceGroupName, id.GrafanaName)
```